thumbor/engines/pil.py

```diff
@@ -81,7 +81,7 @@ def create_image(self, buffer):
         self.exif = img.info.get("exif")
         self.original_mode = img.mode
 
-        if getattr(img, "layers", None):
+        if getattr(img, "layers", None) and getattr(img, "layer", None):
```

Comment:
Differently form `layers`, which can be `0` and therefore rules out sub-sampling ([PIL/Jpeg2KImagePlugin.py](https://github.com/python-pillow/Pillow/blob/243402e78e2bf2b7af478c6d891816e372b5c3f9/src/PIL/Jpeg2KImagePlugin.py#L226)):
```py
        self.layers = 0
```

`layer` is either defined or not ([PIL/JpegImagePlugin.py](https://github.com/python-pillow/Pillow/blob/243402e78e2bf2b7af478c6d891816e372b5c3f9/src/PIL/JpegImagePlugin.py#L354-L358)):
```py
        # Create attributes
        self.bits = self.layers = 0


        # JPEG specifics (internal)
        self.layer = []
```

So, this line can be replaced:

```suggestion
        if hasattr(img, "layer"):
```

Reply:
Done!